crates/biome_js_analyze/src/lint/suspicious/no_double_equals.rs

```diff
@@ -117,6 +153,28 @@ impl Rule for NoDoubleEquals {
     }
 }
 
+/// Rule's options
+#[derive(Clone, Debug, Deserializable, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct NoDoubleEqualsOptions {
+    /// If `true`, an exception is made when comparing with `null`, as it's often relied on to check
+    /// both for `null` or `undefined`.
+    ///
+    /// If `false`, no such exception will be made.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub ignore_null: bool,
+}
+
+impl Default for NoDoubleEqualsOptions {
+    fn default() -> Self {
+        Self { ignore_null: true }
+    }
+}
+fn is_default<T: Default + Eq>(value: &T) -> bool {
+    value == &T::default()
+}
```

Comment:
I suspect there's an issue here, because the default of `bool` is `false`, but for the `ignore_null` field we want it to be `true`, as you `Default` implementation correctly shows. But the `is_default()` function uses a generic, which looks like it will apply to the `bool` type, and so will Serde's `default` annotation.

This will cause a semantic mismatch between the situation when the options aren't provided at all, in which case `NoDoubleEqualsOptions::default()` will be used, and when an empty options object is provided, in which case `ignore_null` will be initialized with the default value for booleans, which is `false`.

I think it's better to be explicit here instead of using a generic, since this will also make the intention more clear:

```suggestion
    #[serde(default = "ignore_null_default", skip_serializing_if = "is_ignore_null_default")]
    pub ignore_null: bool,
}

impl Default for NoDoubleEqualsOptions {
    fn default() -> Self {
        Self { ignore_null: ignore_null_default() }
    }
}

fn ignore_null_default() -> bool {
    true
}

fn is_ignore_null_default(value: &bool) -> bool {
    value == &ignore_null_default()
}
```

Reply:
You're right, your fix looked good so I wrote it and pushed it.

I mixed it up after looking at how `useNamingConvention` was defining its options (especially around the derive macros wanted etc.) and dumbly assumed that the `default` mentions passed to `serde` there somehow led to relying on the corresponding attribute in the full default struct, instead of just the default of a boolean value...
My bad and good catch.

At least by being more explicit like this, the intention is much clearer.